contracts/FlowIDTableStaking.cdc

```diff
@@ -1710,16 +1710,35 @@ pub contract FlowIDTableStaking {
     pub fun getProposedNodeIDs(): [String] {
 
         let nodeIDs = FlowIDTableStaking.getNodeIDs()
+        let approvedNodeIDs: {String: Bool} = FlowIDTableStaking.getApprovedList()
         let proposedNodeIDs: {String: Bool} = {}
 
         for nodeID in nodeIDs {
 
             let nodeRecord = FlowIDTableStaking.borrowNodeRecord(nodeID)
 
-            // To be considered proposed, a node has to have tokens staked + committed equal or above the minimum
-            if self.isGreaterThanMinimumForRole(numTokens: self.NodeInfo(nodeID: nodeRecord.id).totalCommittedWithoutDelegators(), role: nodeRecord.role)
-            {
+            let totalTokensCommitted = nodeRecord.nodeFullCommittedBalance()
+
+            let greaterThanMin = FlowIDTableStaking.isGreaterThanMinimumForRole(numTokens: totalTokensCommitted, role: nodeRecord.role)
+            let nodeIsApproved: Bool =  approvedNodeIDs[nodeID] != nil
```

Comment:
can `approvedNodeIDs[nodeID]`  ever be `false` (instead of `nil`)?

Reply:
not if we use the transactions properly, but I'll change it to `approvedNodeIDs[nodeID] ?? false` so it'll count as false if the value is also false